### ISSUE TITLE
Not available on CRAN

### ISSUE THREAD

**Issue by slhck** [drsimonj/corrr#69]
When I visit 

https://cran.r-project.org/web/packages/corrr/index.html

I get:

> Package ‘corrr’ was removed from the CRAN repository.
> 
> Formerly available versions can be obtained from the archive.
> 
> Archived on 2019-01-20 as check problems were not corrected in time.

What happened there?

**Comment by jsta** [comment]
Looks like an issue with a failing test on Windows:
https://cran.r-project.org/web/checks/check_results_corrr.html

**Comment by edgararuiz** [comment]
I tried fixing that issue when working on a PR for something different and was totally stumped.  It has to do with how `diag()` and `rownames()` interact.  I recall that the fixes I tried worked in some cases and not others.

**Comment by richierocks** [comment]
Is there a plan to get this back onto CRAN? This package is used by DataCamp courses, and I need to know whether we need to stop teaching it.

**Comment by edgararuiz** [comment]
Ok, I think I zoned into the issue and submitted a PR that would love if you can test before I merge it: `devtools::install_github("drsimonj/corrr", devtools::github_pull(71))`

**Comment by edgararuiz** [comment]
Additional info - The PR focuses solely on making sure that the current tests pass, and that there are no warnings when checking the package.  So, a successful test of the version of the package in the PR is to run it through some of your existing code, and it should run without an error and also returns the same values as before.

**Comment by slhck** [comment]
I don't have any of my own tests, but I can confirm that installation works.

Compared to the data shown in the README, I do get different results though:

```
# Simulate three columns correlating about .7 with each other
mu <- rep(0, 3)
Sigma <- matrix(.7, nrow = 3, ncol = 3) + diag(3)*.3
seven <- mvrnorm(n = 1000, mu = mu, Sigma = Sigma)

# Simulate three columns correlating about .4 with each other
mu <- rep(0, 3)
Sigma <- matrix(.4, nrow = 3, ncol = 3) + diag(3)*.6
four <- mvrnorm(n = 1000, mu = mu, Sigma = Sigma)

# Bind together
d <- cbind(seven, four)
colnames(d) <- paste0("v", 1:ncol(d))

# Insert some missing values
d[sample(1:nrow(d), 100, replace = TRUE), 1] <- NA
d[sample(1:nrow(d), 200, replace = TRUE), 5] <- NA

# Correlate
x <- correlate(d)
class(x)

> x
# A tibble: 6 x 7
  rowname        v1        v2        v3        v4       v5       v6
  <chr>       <dbl>     <dbl>     <dbl>     <dbl>    <dbl>    <dbl>
1 v1       NA         0.704     0.714     0.00217  0.0273   -0.0302
2 v2        0.704    NA         0.689    -0.0243   0.00705  -0.0168
3 v3        0.714     0.689    NA        -0.0183   0.00218  -0.0355
4 v4        0.00217  -0.0243   -0.0183   NA        0.428     0.441 
5 v5        0.0273    0.00705   0.00218   0.428   NA         0.421 
6 v6       -0.0302   -0.0168   -0.0355    0.441    0.421    NA
```

**Comment by jsta** [comment]
I'm getting results that match the README...

``` r
library(MASS)
library(corrr)
set.seed(1)

# Simulate three columns correlating about .7 with each other
mu <- rep(0, 3)
Sigma <- matrix(0.7, nrow = 3, ncol = 3) + diag(3) * 0.3
seven <- mvrnorm(n = 1000, mu = mu, Sigma = Sigma)

# Simulate three columns correlating about .4 with each other
mu <- rep(0, 3)
Sigma <- matrix(0.4, nrow = 3, ncol = 3) + diag(3) * 0.6
four <- mvrnorm(n = 1000, mu = mu, Sigma = Sigma)

# Bind together
d <- cbind(seven, four)
colnames(d) <- paste0("v", 1:ncol(d))

# Insert some missing values
d[sample(1:nrow(d), 100, replace = TRUE), 1] <- NA
d[sample(1:nrow(d), 200, replace = TRUE), 5] <- NA

# Correlate
x <- correlate(d)
#> 
#> Correlation method: 'pearson'
#> Missing treated using: 'pairwise.complete.obs'
class(x)
#> [1] "cor_df"     "tbl_df"     "tbl"        "data.frame"
x
#> # A tibble: 6 x 7
#>   rowname         v1        v2        v3         v4       v5       v6
#>   <chr>        <dbl>     <dbl>     <dbl>      <dbl>    <dbl>    <dbl>
#> 1 v1       NA          0.710     0.709     0.000195  0.0214   -0.0435
#> 2 v2        0.710     NA         0.697    -0.0133    0.00928  -0.0338
#> 3 v3        0.709      0.697    NA        -0.0253    0.00109  -0.0201
#> 4 v4        0.000195  -0.0133   -0.0253   NA         0.421     0.442 
#> 5 v5        0.0214     0.00928   0.00109   0.421    NA         0.425 
#> 6 v6       -0.0435    -0.0338   -0.0201    0.442     0.425    NA
```

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> ─ Session info ──────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.5.2 (2018-12-20)
#>  os       Ubuntu 18.04.2 LTS          
#>  system   x86_64, linux-gnu           
#>  ui       X11                         
#>  language (EN)                        
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       America/Toronto             
#>  date     2019-02-27                  
#> 
#> ─ Packages ──────────────────────────────────────────────────────────────
#>  package     * version    date       lib source                           
#>  assertthat    0.2.0      2017-04-11 [1] CRAN (R 3.5.0)                   
#>  backports     1.1.3      2018-12-14 [1] CRAN (R 3.5.1)                   
#>  callr         3.1.1      2018-12-21 [1] CRAN (R 3.5.1)                   
#>  cli           1.0.1      2018-09-25 [1] CRAN (R 3.5.1)                   
#>  colorspace    1.4-0      2018-06-11 [1] R-Forge (R 3.5.0)                
#>  corrr       * 0.3.0.9000 2019-02-27 [1] Github (edgararuiz/corrr@61de052)
#>  crayon        1.3.4      2017-09-16 [1] CRAN (R 3.5.0)                   
#>  desc          1.2.0      2019-02-20 [1] Github (r-lib/desc@42b9578)      
#>  devtools      2.0.1      2018-10-26 [1] CRAN (R 3.5.2)                   
#>  digest        0.6.18     2018-10-10 [1] CRAN (R 3.5.1)                   
#>  dplyr         0.8.0.1    2019-02-15 [1] CRAN (R 3.5.2)                   
#>  evaluate      0.13       2019-02-12 [1] CRAN (R 3.5.2)                   
#>  fansi         0.4.0      2018-10-05 [1] CRAN (R 3.5.1)                   
#>  formatR       1.5        2017-04-25 [1] CRAN (R 3.5.0)                   
#>  fs            1.2.6      2018-08-23 [1] CRAN (R 3.5.1)                   
#>  ggplot2       3.1.0      2018-10-25 [1] CRAN (R 3.5.1)                   
#>  glue          1.3.0      2018-07-17 [1] CRAN (R 3.5.1)                   
#>  gtable        0.2.0      2016-02-26 [1] CRAN (R 3.5.0)                   
#>  highr         0.7        2018-06-09 [1] CRAN (R 3.5.1)                   
#>  htmltools     0.3.6      2017-04-28 [1] CRAN (R 3.5.0)                   
#>  knitr         1.21       2018-12-10 [1] CRAN (R 3.5.1)                   
#>  lazyeval      0.2.1      2017-10-29 [1] CRAN (R 3.5.0)                   
#>  magrittr      1.5        2014-11-22 [1] CRAN (R 3.5.0)                   
#>  MASS        * 7.3-51.1   2018-11-01 [4] CRAN (R 3.5.1)                   
#>  memoise       1.1.0      2017-04-21 [1] CRAN (R 3.5.0)                   
#>  munsell       0.5.0      2018-06-12 [1] CRAN (R 3.5.0)                   
#>  pillar        1.3.1      2018-12-15 [1] CRAN (R 3.5.1)                   
#>  pkgbuild      1.0.2      2018-10-16 [1] CRAN (R 3.5.1)                   
#>  pkgconfig     2.0.2      2018-08-16 [1] CRAN (R 3.5.1)                   
#>  pkgload       1.0.2      2018-10-29 [1] CRAN (R 3.5.1)                   
#>  plyr          1.8.4      2016-06-08 [1] CRAN (R 3.5.0)                   
#>  prettyunits   1.0.2      2015-07-13 [1] CRAN (R 3.5.0)                   
#>  processx      3.2.1      2018-12-05 [1] CRAN (R 3.5.1)                   
#>  ps            1.3.0      2018-12-21 [1] CRAN (R 3.5.1)                   
#>  purrr         0.3.0      2019-01-27 [1] CRAN (R 3.5.2)                   
#>  R6            2.4.0      2019-02-14 [1] CRAN (R 3.5.2)                   
#>  Rcpp          1.0.0      2018-11-07 [1] CRAN (R 3.5.1)                   
#>  remotes       2.0.2      2018-10-30 [1] CRAN (R 3.5.1)                   
#>  rlang         0.3.1      2019-01-08 [1] CRAN (R 3.5.2)                   
#>  rmarkdown     1.11       2018-12-08 [1] CRAN (R 3.5.1)                   
#>  rprojroot     1.3-2      2018-01-03 [1] CRAN (R 3.5.0)                   
#>  scales        1.0.0.9000 2019-02-16 [1] Github (hadley/scales@c374014)   
#>  sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 3.5.1)                   
#>  stringi       1.3.1      2019-02-13 [1] CRAN (R 3.5.2)                   
#>  stringr       1.4.0      2019-02-10 [1] CRAN (R 3.5.2)                   
#>  testthat      2.0.1      2018-10-13 [1] CRAN (R 3.5.2)                   
#>  tibble        2.0.1      2019-01-12 [1] CRAN (R 3.5.2)                   
#>  tidyselect    0.2.5      2018-10-11 [1] CRAN (R 3.5.1)                   
#>  usethis       1.4.0.9000 2018-11-21 [1] Github (krlmlr/usethis@290a9e6)  
#>  utf8          1.1.4      2018-05-24 [1] CRAN (R 3.5.1)                   
#>  withr         2.1.2      2018-03-15 [1] CRAN (R 3.5.0)                   
#>  xfun          0.4        2018-10-23 [1] CRAN (R 3.5.1)                   
#>  yaml          2.2.0      2018-07-25 [1] CRAN (R 3.5.1)                   
#> 
#> [1] /home/jose/R/x86_64-pc-linux-gnu-library/3.5
#> [2] /usr/local/lib/R/site-library
#> [3] /usr/lib/R/site-library
#> [4] /usr/lib/R/library
```

</details>

**Comment by edgararuiz** [comment]
Good idea to try running the code in README.  I tried it out and got the same results as the one currently in the master.  I'm sorry @slhck , not sure why you're not getting the same results, can you share your session info with us?

**Comment by slhck** [comment]
Here's the entire log:

```
R version 3.5.2 (2018-12-20) -- "Eggshell Igloo"
Copyright (C) 2018 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin17.7.0 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> packrat::off()
Packrat mode off. Resetting library paths to:
- "/Users/werner/Library/R/3.5/library"
- "/usr/local/Cellar/r/3.5.2_2/lib/R/library"
> devtools::install_github("drsimonj/corrr", devtools::github_pull(71))
Skipping install of 'corrr' from a github remote, the SHA1 (0c269a2a) has not changed since last install.
  Use `force = TRUE` to force installation
> library(MASS)
> library(corrr)
> set.seed(1)
> 
> # Simulate three columns correlating about .7 with each other
> mu <- rep(0, 3)
> Sigma <- matrix(.7, nrow = 3, ncol = 3) + diag(3)*.3
> seven <- mvrnorm(n = 1000, mu = mu, Sigma = Sigma)
> 
> # Simulate three columns correlating about .4 with each other
> mu <- rep(0, 3)
> Sigma <- matrix(.4, nrow = 3, ncol = 3) + diag(3)*.6
> four <- mvrnorm(n = 1000, mu = mu, Sigma = Sigma)
> 
> # Bind together
> d <- cbind(seven, four)
> colnames(d) <- paste0("v", 1:ncol(d))
> 
> # Insert some missing values
> d[sample(1:nrow(d), 100, replace = TRUE), 1] <- NA
> d[sample(1:nrow(d), 200, replace = TRUE), 5] <- NA
> 
> # Correlate
> x <- correlate(d)

Correlation method: 'pearson'
Missing treated using: 'pairwise.complete.obs'

> class(x)
[1] "cor_df"     "tbl_df"     "tbl"        "data.frame"
> x
# A tibble: 6 x 7
  rowname        v1        v2        v3        v4       v5       v6
  <chr>       <dbl>     <dbl>     <dbl>     <dbl>    <dbl>    <dbl>
1 v1       NA         0.704     0.714     0.00217  0.0273   -0.0302
2 v2        0.704    NA         0.689    -0.0243   0.00705  -0.0168
3 v3        0.714     0.689    NA        -0.0183   0.00218  -0.0355
4 v4        0.00217  -0.0243   -0.0183   NA        0.428     0.441 
5 v5        0.0273    0.00705   0.00218   0.428   NA         0.421 
6 v6       -0.0302   -0.0168   -0.0355    0.441    0.421    NA
```

<details>
<summary>Session info:</summary>
<pre>
> devtools::session_info()
Session info ------------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.5.2 (2018-12-20)
 system   x86_64, darwin17.7.0        
 ui       RStudio (1.1.463)           
 language (EN)                        
 collate  en_US.UTF-8                 
 tz       Europe/Vienna               
 date     2019-02-27                  

Packages ----------------------------------------------------------------------------------------------------------
 package    * version    date       source                          
 assertthat   0.2.0      2017-04-11 CRAN (R 3.5.0)                  
 base       * 3.5.2      2019-01-13 local                           
 cli          1.0.1      2018-09-25 cran (@1.0.1)                   
 colorspace   1.4-0      2019-01-13 cran (@1.4-0)                   
 compiler     3.5.2      2019-01-13 local                           
 corrr      * 0.3.0.9000 2019-02-27 Github (drsimonj/corrr@0c269a2) 
 crayon       1.3.4      2017-09-16 CRAN (R 3.5.0)                  
 curl         3.2        2018-03-28 CRAN (R 3.5.0)                  
 datasets   * 3.5.2      2019-01-13 local                           
 devtools     1.13.5     2018-02-18 CRAN (R 3.5.0)                  
 digest       0.6.18     2018-10-10 CRAN (R 3.5.2)                  
 dplyr        0.8.0.1    2019-02-15 cran (@0.8.0.1)                 
 evaluate     0.12       2018-10-09 CRAN (R 3.5.2)                  
 fansi        0.4.0      2018-10-05 cran (@0.4.0)                   
 ggplot2      3.1.0      2018-10-25 cran (@3.1.0)                   
 git2r        0.21.0     2018-01-04 CRAN (R 3.5.0)                  
 glue         1.3.0      2018-07-17 cran (@1.3.0)                   
 graphics   * 3.5.2      2019-01-13 local                           
 grDevices  * 3.5.2      2019-01-13 local                           
 grid         3.5.2      2019-01-13 local                           
 gtable       0.2.0      2016-02-26 CRAN (R 3.5.0)                  
 htmltools    0.3.6      2017-04-28 CRAN (R 3.5.2)                  
 httr         1.3.1      2017-08-20 CRAN (R 3.5.0)                  
 knitr        1.21       2018-12-10 CRAN (R 3.5.2)                  
 lazyeval     0.2.1      2017-10-29 CRAN (R 3.5.0)                  
 magrittr     1.5        2014-11-22 CRAN (R 3.5.2)                  
 MASS       * 7.3-51.1   2018-11-01 CRAN (R 3.5.2)                  
 memoise      1.1.0      2017-04-21 CRAN (R 3.5.0)                  
 methods    * 3.5.2      2019-01-13 local                           
 munsell      0.5.0      2018-06-12 cran (@0.5.0)                   
 packrat      0.5.0-5    2019-02-20 Github (rstudio/packrat@9dee5d8)
 pillar       1.3.1      2018-12-15 cran (@1.3.1)                   
 pkgconfig    2.0.2      2018-08-16 cran (@2.0.2)                   
 plyr         1.8.4      2016-06-08 CRAN (R 3.5.0)                  
 purrr        0.3.0      2019-01-27 cran (@0.3.0)                   
 R6           2.4.0      2019-02-14 cran (@2.4.0)                   
 Rcpp         1.0.0      2018-11-07 CRAN (R 3.5.2)                  
 rlang        0.3.1      2019-01-08 cran (@0.3.1)                   
 rmarkdown    1.11       2018-12-08 CRAN (R 3.5.2)                  
 rstudioapi   0.7        2017-09-07 CRAN (R 3.5.0)                  
 scales       1.0.0      2018-08-09 cran (@1.0.0)                   
 stats      * 3.5.2      2019-01-13 local                           
 stringi      1.2.4      2018-07-20 CRAN (R 3.5.2)                  
 stringr      1.3.1      2018-05-10 CRAN (R 3.5.2)                  
 tibble       2.0.1      2019-01-12 cran (@2.0.1)                   
 tidyselect   0.2.5      2018-10-11 cran (@0.2.5)                   
 tools        3.5.2      2019-01-13 local                           
 utf8         1.1.4      2018-05-24 cran (@1.1.4)                   
 utils      * 3.5.2      2019-01-13 local                           
 withr        2.1.2      2018-03-15 CRAN (R 3.5.0)                  
 xfun         0.4        2018-10-23 CRAN (R 3.5.2)                  
 yaml         2.2.0      2018-07-25 CRAN (R 3.5.2) 
</pre>
</details>

**Comment by slhck** [comment]
I notice I got 0c269a2 while you have 61de052 … did I do something wrong while installing?

**Comment by edgararuiz** [comment]
Can we try again but with forcing the update? It is 61de052 the correct commit

**Comment by slhck** [comment]
I still get the wrong version:

```
> devtools::install_github("drsimonj/corrr", devtools::github_pull(71), force = TRUE)
Downloading GitHub repo drsimonj/corrr@master
from URL https://api.github.com/repos/drsimonj/corrr/zipball/master
Installing corrr
'/usr/local/Cellar/r/3.5.2_2/lib/R/bin/R' --no-site-file --no-environ --no-save --no-restore --quiet CMD INSTALL  \
  '/private/var/folders/1s/9jbdrzg17wvcnr76p_ntdnbw0000gn/T/RtmpgtuKS7/devtools1821b6ad5c03f/drsimonj-corrr-0c269a2'  \
  --library='/Users/werner/Library/R/3.5/library' --install-tests 
```

I actually don't see your commit when browsing https://github.com/drsimonj/corrr/commits/master, and there are also no other branches.

**Comment by jsta** [comment]
I wonder if you are having a `devtools` issue. It looks like your version is somewhat old.

**Comment by slhck** [comment]
You're right, good observation. I hadn't done an update in a while. Updated all packages – now everything is matching.

**Comment by edgararuiz** [comment]
The package is now back in CRAN 
Fixed by 6ab74a38d3ed70510730ccd3db17ab1dc912678c

**Comment by slhck** [comment]
Thank you!
